runtime/vm/resolvesupport.cpp

```diff
@@ -941,20 +966,21 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 			}
 		
 			if (ramCPEntry != NULL) {
-				UDATA valueOffset = fieldOffset;
+				UDATA valueOffset = fieldOffset;				
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
-				if ('Q' == *J9UTF8_DATA(signature)) {
-					J9FlattenedClassCache *flattenedClassCache = classFromCP->flattenedClassCache;
-					J9Class *flattenableClass = NULL;
-					UDATA index = findIndexInFlattenedClassCache(flattenedClassCache, nameAndSig);
-					flattenableClass = J9_VM_FCC_ENTRY_FROM_FCC(flattenedClassCache, index)->clazz;
-
+				if ('Q' == J9UTF8_DATA(signature)[0]) {
+					J9FlattenedClassCache *flattenedClassCache;
```

Comment:
flattenedClassCache can be defined in a higher scope since it is used here, and earlier in the function.